engine/proofview.ml

```diff
@@ -588,6 +588,13 @@ let shelve =
   InfoL.leaf (Info.Tactic (fun () -> Pp.str"shelve")) >>
   Shelf.modify (fun gls -> gls @ initial)
 
+let shelve_goals l =
+  let open Proof in
+  Comb.get >>= fun initial ->
+  let comb = CList.filter (fun g -> not (CList.mem g l)) initial in
+  Comb.set comb >>
+  InfoL.leaf (Info.Tactic (fun () -> Pp.str"shelve_goals")) >>
+  Shelf.modify (fun gls -> gls @ l)
```

Comment:
Ah, I see what you mean now, it should check that all the goals that are shelved are in the focus indeed. I'll work on that.